rig/machine_control/machine_controller.py

```diff
@@ -373,6 +377,49 @@ def read_struct_field(self, struct_name, field_name, x, y, p=0):
             return unpacked
 
     @ContextMixin.use_contextual_arguments()
+    def write_struct_field(self, struct_name, field_name, values, x, y, p):
+        """Write a value into a struct.
+
+        This method is particularly useful for writing values into the ``sv``
+        struct which contains some configuration data.  See ``sark.h`` for
+        details.
+
+        Parameters
+        ----------
+        struct_name : string
+            Name of the struct to write to, e.g., `"sv"`
+        field_name : string
+            Name of the field to write, e.g., `"random"`
```

Comment:
Awesome, I didn't know about that field! /me should really brows the sv struct more...